examples/save_plot/src/main.rs

```diff
@@ -0,0 +1,144 @@
+use eframe::egui::ColorImage;
+
+fn main() -> Result<(), eframe::Error> {
+    // Log to stdout (if you run with `RUST_LOG=debug`).
+    tracing_subscriber::fmt::init();
+
+    let options = eframe::NativeOptions {
+        initial_window_size: Some(egui::vec2(350.0, 400.0)),
+        ..Default::default()
+    };
+    eframe::run_native(
+        "My egui App with a plot",
+        options,
+        Box::new(|_cc| Box::new(MyApp::default())),
+    )
+}
+
+struct MyApp {
+    /// Unit: fraction of full screen
+    plot_location: egui::Rect,
+    screenshot: Option<ColorImage>,
+}
+
+impl Default for MyApp {
+    fn default() -> Self {
+        Self {
+            plot_location: egui::Rect::EVERYTHING,
+            screenshot: None,
+        }
+    }
+}
+
+impl eframe::App for MyApp {
+    fn update(&mut self, ctx: &egui::Context, frame: &mut eframe::Frame) {
+        egui::CentralPanel::default().show(ctx, |ui| {
+            // these are just some dummy variables for the example
+            let height = 200.0;
+            let border_x = 11.0;
+            let border_y = 18.0;
+            let width = 300.0;
+
+            // get the size of the window
+            let window_width = ui.available_size().x;
+            let window_height = ui.available_size().y;
+
+            ui.heading("My egui Application");
+
+            // add some whitespace in y direction
+            ui.add_space(border_y);
+
+            if ui.button("Save Plot").clicked() {
+                frame.request_screenshot();
+            }
+
+            // add some whitespace in y direction
+            ui.add_space(border_y);
+
+            // this needs to be outside of the ui.horizontal()
+            let plot_location_y = window_height - ui.available_size().y;
+            ui.horizontal(|ui| {
+                // add some whitespace in x direction
+                ui.add_space(border_x);
+
+                // obviously this needs to be after the last ui.add_space
+                let plot_location_x = window_width - ui.available_size().x;
+
+                // lets set the relative plot location for plot saving purposes
+                self.plot_location = egui::Rect::from_two_pos(
+                    egui::Pos2 {
+                        x: plot_location_x / window_width,
+                        y: plot_location_y / window_height,
+                    },
+                    egui::Pos2 {
+                        x: (plot_location_x + width) / window_width,
+                        y: (plot_location_y + height) / window_height,
+                    },
+                );
+                let my_plot = Plot::new("My Plot")
+                    .height(height)
+                    .width(width)
+                    .legend(Legend::default());
+
+                // let's create a dummy line in the plot
+                let graph: Vec<[f64; 2]> = vec![[0.0, 1.0], [2.0, 3.0], [3.0, 2.0]];
+                my_plot.show(ui, |plot_ui| {
```

Comment:
`.show` returns an [`InnerResponse`](https://docs.rs/egui/latest/egui/struct.InnerResponse.html) containing a [`Response`](https://docs.rs/egui/latest/egui/struct.Response.html) which has the `Rect` that the plot covered. You should be able to use that for initializing your `plot_location`

Reply:
it also seems easiest to store the location in ui points (as returned by egui) and then pass in `pixels_per_point` to `ColorImage::region` so you don't need to do your own normalization of coordinates

Reply:
Thanks!!! That's what I was missing, now it's much cleaner!